drivers/adodb-sqlite3.inc.php

```diff
@@ -596,6 +596,119 @@ function year($fld)
 		$x = "strftime('%Y',$fld)";
 		return $x;
 	}
+	
+	/**
+	* SQLite update for blob 
+	*
+	* SQLite must be a fully prepared statement, so where can either
+	* be an array (array params) or a string that we will
+	* do our best to unpack and turn ito a prepared statement
+	*
+	* @param string $table
+	* @param string $column
+	* @param string $val
+	* @param mixed  $where
+	* @param string $blobtype ignored
+	*
+	* @return success 
+	*/
+	function updateBlob($table,$column,$val,$where,$blobtype='BLOB') 
+	{
+		if (is_array($where))
+		{
+			/*
+			* We were passed a key=>value pair set
+			*/
+			$params = $where;
+		}
+		else
+		{
+			/*
+			* We have to disasemble the statements into keys
+			* and values
+			*/
+			$params = array();
+			$temp   = preg_split('/(where|and)+/i',$where);
+			$where  = array_filter($temp);
+			
+			foreach($where as $wValue)
+			{
+				$wTemp = preg_split('/[= \']+/',$wValue);
+				$wTemp = array_filter($wTemp);
+				$wTemp = array_values($wTemp);
+				$params[$wTemp[0]] = $wTemp[1];
+			}
+		}
+		$paramWhere = array();
+		
+		foreach ($params as $bindKey=>$bindValue) 
+		{
+			$paramWhere[] = $bindKey . '=?';
+		}
+		
+		$where = 'WHERE ' . implode(' AND ' , $paramWhere);
+		
+		$sql = "UPDATE $table SET $column=? $where";
+		
+		/*
+		* Prepare the statement
+		*/
+		$stmt = $this->_connectionID->prepare($sql);
+		
+		/*
+		* Set the first bind value equal to value we want to update
+		*/
+		$bindOk = $stmt->bindValue(1, $val, SQLITE3_BLOB);
+		
+		/*
+		* Build as many keys as available
+		*/
+		$bindIndex = 2;
+		foreach ($params as $bindKey=>$bindValue) 
+		{
+			
+			if (is_integer($v) || is_bool($v) || is_float($v))
+				$bindOk = $stmt->bindValue($bindIndex, $bindValue, SQLITE3_NUM);
+					
+			else if(is_object($v))
+				/*
+				* Assume a blob
+				*/
+				$bindOk = $stmt->bindValue($bindIndex, $bindValue, SQLITE3_NUM);
+			else
+				$bindOk = $stmt->bindValue($bindIndex, $bindValue, SQLITE3_TEXT);
+				
+			$bindIndex++;
+		}
+		
+		/*
+		* Now execute the update. NB this is SQLite execute,
+		* not ADOdb
+		*/
+		$ok = $stmt->execute();
+		return is_object($ok) ? true : false;
```

Comment:
The `? true : false` bit is unnecessary, is_object already returns bool - just `return is_object($ok)`